### PR TITLE
Add possibility to append records to dns

### PR DESCRIPTION
**Pull request by artheus**
Eg. SRV records usually needs more than one single
record with the same name.

Suggested change based on my issue #28993

##### SUMMARY
Suggessted fix to #28993

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
nsupdate

##### ANSIBLE VERSION
N/A

##### ADDITIONAL INFORMATION
See issue #28993

### CODE REVIEW

**Comment by artheus** [comment]
@abadger @gundalow Please review.

File: lib/ansible/modules/net_tools/nsupdate.py [hunk 1]
```diff
@@ -193,15 +209,15 @@ def __do_update(self, update):
             self.module.fail_json(msg='DNS server error: (%s): %s' % (e.__class__.__name__, to_native(e)))
         return response
 
-    def create_or_update_record(self):
+    def create_or_update_record(self, state, allow_duplicates):
```
Comment (abadger) [inline lib/ansible/modules/net_tools/nsupdate.py:212]: It doesn't look like state is needed.

File: lib/ansible/modules/net_tools/nsupdate.py [hunk 2]
```diff
@@ -193,15 +209,15 @@ def __do_update(self, update):
...
         result = {'changed': False, 'failed': False}
 
         exists = self.record_exists()
         if exists in [0, 2]:
             if self.module.check_mode:
                 self.module.exit_json(changed=True)
 
-            if exists == 0:
+            if exists == 0 or allow_duplicates:
```
Comment (abadger) [inline lib/ansible/modules/net_tools/nsupdate.py:220]: @bcoca.. I don't know DNS well.  Is this all that's needed or do we have to check whether the record would be exactly the same as an existing record and not create it if so?

Comment (bcoca) [reply lib/ansible/modules/net_tools/nsupdate.py:220]: depends ... some DNS servers allow multiple entry as a way to skew the load, i.e:
```
www.example.com    10.10.1.1
www.example.com    10.10.1.1
www.example.com    10.10.1.2
```
^ will send 2/3 of traffic to first IP and 1/3 to the last .. so technically it should be accepted .. but that depends on the server software (some do this, some ignore, some error out).

Comment (nerzhul) [reply lib/ansible/modules/net_tools/nsupdate.py:220]: dupe are not standard, when you want to do load balancing with DNS you should use SRV records, or have a load balancer

Comment (bcoca) [reply lib/ansible/modules/net_tools/nsupdate.py:220]: @nerzhul this 'feature' predates the RFC that introduced SRV and was used as a 'cheap' load balancing trick

Comment (nerzhul) [reply lib/ansible/modules/net_tools/nsupdate.py:220]: i really don't like a such crap in DNS nsupdate should take care about all DNS implementation , and bind9 is the reference and doesn't support a such thing

**Comment by artheus** [comment]
@abadger @nerzhul @bcoca Hi, how does it look on merging this?

File: lib/ansible/modules/net_tools/nsupdate.py (hunk 2 above)
Comment (artheus) [reply lib/ansible/modules/net_tools/nsupdate.py:220]: The reason for me to want to add this is that I wanted to add multiple SRV records. There are also some other records that are, by default, supported to have multiple entries.
So one way is to allow users to add multiple of everything, which should be safe as long as they know what they're doing. But can be quite bad if someone who don't really know uses this feature.
But as we add this as an explicit flag which needs to be set, I think that by itself should trigger some thoughts in users.

I see three alternatives.
1. Be strict about this and only allow multiple records for the ones that allow it by default.
2. Same as (1) but also add more flags for special cases such as "DNS Load balancing"
3. Keep as is, and hope people know what they are doing.

**Comment by DanyC97** [comment]
@artheus hi, are you still looking to get this PR merged ? 
i'm in same possition where i need to get multiple SRV records 

```
_etcd-server-ssl._tcp..<base_domain>  86400 IN    SRV 0        10     2380 etcd-0.<base_domain>.
_etcd-server-ssl._tcp.<base_domain>   86400 IN    SRV 0        10     2380 etcd-1.<base_domain>.
_etcd-server-ssl._tcp.<base_domain>   86400 IN    SRV 0        10     2380 etcd-2.<base_domain>.
```
and to my surprise i only get the last entry..hmm

**Comment by madkiss** [comment]
Can we please get this in? :(

**Comment by artheus** [comment]
> @artheus hi, are you still looking to get this PR merged ?
> i'm in same possition where i need to get multiple SRV records
> 
> ```
> _etcd-server-ssl._tcp.<base_domain>  86400 IN    SRV 0        10     2380 etcd-0.<base_domain>.
> _etcd-server-ssl._tcp.<base_domain>   86400 IN    SRV 0        10     2380 etcd-1.<base_domain>.
> _etcd-server-ssl._tcp.<base_domain>   86400 IN    SRV 0        10     2380 etcd-2.<base_domain>.
> ```
> 
> and to my surprise i only get the last entry..hmm


> Can we please get this in? :(

I would really like to see this get merged some time. But I am not in strong need of it getting merged, as I override the module locally in my ansible main repo.

**Comment by artheus** [comment]
From what I have seen on this PR, I would suspect that the best option might be to override this module, or write a separate module to get this behaviour in your ansible setup.

**Pull request closed without merging**